### PR TITLE
Add XML documentation for publishing library to .Net API Reference

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Lots of hardwork coming in!

Contributors:
@JRahnama @v-kaywon @cheenamalhotra 

Changes with this PR:
- Adds common XML doc snippets to be extracted for XML documentation
- Adds Samples to repository included in documentation
- Updated both NetFx and NetCore drivers with documentation includes.
- GitIgnore update to exclude `nuget.exe`
- build.proj now defaults to `BuildAllConfigurations` target. Addresses #197
- Added include tags in 'ref' classes for Visual Studio Intellisense support.

Additional Changes in this PR:
- `Microsoft.Data.SqlTypes.SqlFileStream` added in NetFx.
- Existing tests run against the newly added SqlFileStream class.

As of now, this PR brings documentation over from System.Data.SqlClient.
Work to be continued to enhance missing docs in future PRs.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.Manual.cs [hunk 1]
```diff
@@ -7,11 +7,13 @@ namespace Microsoft.Data.SqlClient
     public sealed partial class SqlCommand : System.Data.Common.DbCommand
     {
         // SqlCommand expects IDisposable methods to be implemented via System.ComponentModel.Component, which it no longer inherits from
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/Dispose/*'/>
         override protected void Dispose(bool disposing) { }
     }
     public sealed partial class SqlConnection : System.Data.Common.DbConnection
     {
         // SqlConection expects IDisposable methods to be implemented via System.ComponentModel.Component, which it no longer inherits from
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/Dispose/*'/>
```
Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.Manual.cs]:
```suggestion
        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml' path='docs/members[@name="SqlConnection"]/Dispose/*'/>
```

File: src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs [hunk 2]
```diff
@@ -4,1081 +4,1849 @@
...
         public override System.Threading.Tasks.Task<object> ExecuteScalarAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/ExecuteXmlReader[@name="default"]/*'/>
         public System.Xml.XmlReader ExecuteXmlReader() { throw null; }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/ExecuteXmlReaderAsync[@name="default"]/*'/>
         public System.Threading.Tasks.Task<System.Xml.XmlReader> ExecuteXmlReaderAsync() { throw null; }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/ExecuteXmlReaderAsync[@name="CancellationToken"]/*'/>
         public System.Threading.Tasks.Task<System.Xml.XmlReader> ExecuteXmlReaderAsync(System.Threading.CancellationToken cancellationToken) { throw null; }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/SqlCommand/*'/>
```
Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs]:
```suggestion
        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/Prepare/*'/>
```

File: src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs [hunk 3]
```diff
@@ -4,1081 +4,1849 @@
...
         [System.ComponentModel.BrowsableAttribute(false)]
         [System.ComponentModel.DesignerSerializationVisibilityAttribute(0)]
         public override System.Data.ConnectionState State { get { throw null; } }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml' path='docs/members[@name="SqlConnection"]/StatisticsEnabled/*'/>
         [System.ComponentModel.DefaultValueAttribute(false)]
         public bool StatisticsEnabled { get { throw null; } set { } }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml' path='docs/members[@name="SqlConnection"]/WorkstationId/*'/>
         [System.ComponentModel.DesignerSerializationVisibilityAttribute(0)]
         public string WorkstationId { get { throw null; } }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml' path='docs/members[@name="SqlConnection"]/I
```
Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs]:
```suggestion
        /// <include file='..\..\..\..\doc\snippets\Microsoft.Data.SqlClient.DataClassification\SensitivityProperty.xml' path='docs/members[@name="SensitivityProperty"]/ctor/*' />
```

File: src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs [hunk 4]
```diff
@@ -4,1265 +4,2192 @@
...
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/ApplicationIntent.xml' path='docs/members[@name="ApplicationIntent"]/ReadWrite/*'/>
         ReadWrite = 0
     }
+
+    /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/OnChangeEventHandler.xml' path='docs/members[@name="OnChangeEventHandler"]/*'/>
     public delegate void OnChangeEventHandler(object sender, Microsoft.Data.SqlClient.SqlNotificationEventArgs e);
+    /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/PoolBlockingPeriod.xml' path='docs/members[@name="PoolBlockingPeriod"]/PoolBlockingPeriod/*'/>
     public enum PoolBlockingPeriod
     {
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/PoolBlockingPeriod.xml' path='docs/members[@name="PoolBlockingPeriod"]/Auto/*'/>
```
Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs]:
```suggestion
        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/PoolBlockingPeriod.xml' path='docs/members[@name="PoolBlockingPeriod"]/AlwaysBlock/*'/>
```

File: src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs [hunk 5]
```diff
@@ -4,1265 +4,2192 @@
...
         AlwaysBlock = 1,
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/PoolBlockingPeriod.xml' path='docs/members[@name="PoolBlockingPeriod"]/AlwaysBlock/*'/>
```
Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs]:
```suggestion
        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/PoolBlockingPeriod.xml' path='docs/members[@name="PoolBlockingPeriod"]/Auto/*'/>
```

File: src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs [hunk 6]
```diff
@@ -4,1265 +4,2192 @@
...
         public override System.Security.IPermission CreatePermission() { throw null; }
     }
+    /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlColumnEncryptionCertificateStoreProvider.xml' path='docs/members[@name="SqlColumnEncryptionCertificateStoreProvider"]/SqlColumnEncryptionCertificateStoreProvider/*'/>
     public partial class SqlColumnEncryptionCertificateStoreProvider : Microsoft.Data.SqlClient.SqlColumnEncryptionKeyStoreProvider
     {
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlColumnEncryptionCertificateStoreProvider.xml' path='docs/members[@name="SqlColumnEncryptionCertificateStoreProvider"]/SqlColumnEncryptionCertificateStoreProvider/*'/>
         public const string ProviderName = "MSSQL_CERTIFICATE_STORE";
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlColumnEncryptionCertificateStoreProvider.xml' path='docs/members[@name="SqlColumnEncryptionCertificateStoreProvider"]/SqlColumnEncryptionCertificateStoreProvider/*'/>
```
Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs]:
```suggestion
        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlColumnEncryptionCertificateStoreProvider.xml' path='docs/members[@name="SqlColumnEncryptionCertificateStoreProvider"]/ctor/*'/>
```
Similar typo in other functions of this class.

File: src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs [hunk 7]
```diff
@@ -4,1265 +4,2192 @@
...
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/CreateDbParameter/*'/>
         protected override System.Data.Common.DbParameter CreateDbParameter() { throw null; }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/CreateParameter/*'/>
         public new Microsoft.Data.SqlClient.SqlParameter CreateParameter() { throw null; }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/Dispose/*'/>
         protected override void Dispose(bool disposing) { }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/EndExecuteNonQuery[@name="IAsyncResult"]/*'/>
         public int EndExecuteNonQuery(System.IAsyncResult asyncResult) { throw null; }
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/EndExecuteReader[@name="IAsyncResu
```
Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs]:
```suggestion
        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml' path='docs/members[@name="SqlCommand"]/Prepare/*'/>
```

Comment (v-kaywon) [inline src/Microsoft.Data.SqlClient/netfx/ref/Microsoft.Data.SqlClient.cs]:
```suggestion
        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlCommandBuilder.xml' path='docs/members[@name="SqlCommandBuilder"]/QuoteSuffix/*'/>
```

**Comment by cheenamalhotra** [comment]
NuGet Package from Commits Feed:
[Microsoft.Data.SqlClient 1.1.0-build.19274.1-d6f2e29](https://dev.azure.com/sqlclientdrivers-ci/sqlclient/_packaging?_a=package&feed=Microsoft.Data.SqlClient.Commits&package=Microsoft.Data.SqlClient&protocolType=NuGet&version=1.1.0-build.19274.1-d6f2e29&view=overview)

**Pull request merged**